src/Subtitles/index.js

```diff
@@ -0,0 +1,163 @@
+      if (self._captions && self._captions.length) {
+        if (
+          Number(timeIndex.toFixed(0)) <=
+          Number(self._captions[self._captions.length - 1].start.toFixed(0))
+        ) {
+          if (Number(timeIndex.toFixed(0)) < Number(self._captions[0].end.toFixed(0))) {
+            if (self._captions[1] && self._captions[1].payload) {
+              self._nextSubtitle = self._captions[1]
+            }
+            self._currentSubtitle = self._captions[0]
+          } else {
+            for (let i = 0; i < self._captions.length; i++) {
+              if (Number(self._captions[i].start.toFixed(0)) >= Number(timeIndex.toFixed(0))) {
+                self._captions[i + 1] && self._captions[i + 1].payload
+                  ? (self._nextSubtitle = self._captions[i + 1])
+                  : { payload: '' }
+                self._currentSubtitle = self._captions[i]
+                break
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // parses subtitle file and returns list of time, text objects
+  static parseSubtitles(plainSub) {
+    let linesArray = plainSub
+      .trim()
+      .replace('\r\n', '\n')
+      .split(/[\r\n]/)
+      .map(line => {
+        return line.trim()
+      })
+    let cues = []
+    let start = null
+    let end = null
+    let payload = null
+    let lines = linesArray.filter(item => item !== '' && isNaN(item))
+    linesArray = []
+    for (let i = 0; i < lines.length; i++) {
+      if (lines[i].indexOf('-->') >= 0) {
+        let splitted = lines[i].split(/[ \t]+-->[ \t]+/)
+
+        start = Subtitles.parseTimeStamp(splitted[0])
+        end = Subtitles.parseTimeStamp(splitted[1])
+      } else if (lines[i] !== '') {
+        if (start && end) {
+          if (i + 1 < lines.length && lines[i + 1].indexOf('-->') >= 0) {
+            let cue = new SubtitleAsset({
+              start,
+              end,
+              payload: payload ? payload + ' ' + lines[i] : lines[i],
+            })
+            cues.push(cue)
+            start = null
+            end = null
+            payload = null
+          } else {
+            payload = payload ? payload + ' ' + lines[i] : lines[i]
+          }
+        }
+      } else if (start && end) {
+        if (payload == null) {
+          payload = lines[i]
+        } else {
+          payload += ' ' + lines[i]
+        }
+      }
+    }
+    if (start && end) {
+      let match = /<(.*?)>/g
+      if (payload) {
+        payload.replace(match, '')
+      }
+      let cue = new SubtitleAsset({ start, end, payload })
+      cues.push(cue)
+    }
+    this._captions = cues
+    return this._captions
+  }
+
+  // parses timestamp in subtitle file into seconds
+  static parseTimeStamp(s) {
+    let match = s.match(/^(?:([0-9]+):)?([0-5][0-9]):([0-5][0-9](?:[.,][0-9]{0,3})?)/)
+
+    let hours = parseInt(match[1] || '0', 10)
+    let minutes = parseInt(match[2], 10)
+    let seconds = parseFloat(match[3].replace(',', '.'))
```

Comment:
Is this meant to support notations like `03:02:01.215` and `03:02:01,215`? Because when I do a `parseFloat('12.205')` it yields `12.205`, but if I do a `parseFloat(12,205)` it yields `12`, which is I think expected. So, that means I guess that the ',' and '.' should be flipped over in your `.replace` statement